src/Microsoft.Diagnostics.Runtime/src/Linux/LinuxLiveDataReader.cs

```diff
@@ -195,48 +237,19 @@ public unsafe bool GetThreadContext(uint threadID, uint contextFlags, Span<byte>
             if (!_threadIDs.Contains(threadID) || context.Length != AMD64Context.Size)
                 return false;
 
-            fixed (byte* ctxPtr = context)
-            {
-                AMD64Context* ctx = (AMD64Context*)ctxPtr;
-                ctx->ContextFlags = contextFlags;
-                IntPtr ptr = Marshal.AllocHGlobal(sizeof(RegSetX64));
-                try
-                {
-                    ptrace(PTRACE_GETREGS, (int)threadID, IntPtr.Zero, ptr);
-                    RegSetX64 r = Marshal.PtrToStructure<RegSetX64>(ptr);
-                    CopyContext(ctx, ref r);
-                }
-                finally
-                {
-                    Marshal.FreeHGlobal(ptr);
-                }
-            }
-            return true;
-        }
-
-        public unsafe bool GetThreadContext(uint threadID, uint contextFlags, uint contextSize, byte[] context)
-        {
-            this.LoadThreads();
-            if (!_threadIDs.Contains(threadID) || contextSize != AMD64Context.Size)
-            {
-                return false;
-            }
-            IntPtr ptrContext = Marshal.AllocHGlobal(sizeof(AMD64Context));
-            AMD64Context* ctx = (AMD64Context*)ptrContext;
-            ctx->ContextFlags = contextFlags;
-            IntPtr ptr = Marshal.AllocHGlobal(sizeof(RegSetX64));
+            ref AMD64Context ctx = ref Unsafe.As<byte, AMD64Context>(ref MemoryMarshal.GetReference(context));
+            ctx.ContextFlags = contextFlags;
+            IntPtr registerSet = Marshal.AllocHGlobal(sizeof(RegSetX64));
```

Comment:
Can you swap this to ArrayPool.Rent?  (I know it's in the old code, but might as well update it here.